tests/e2e/login_test.go

```diff
@@ -0,0 +1,98 @@
+				t := runCmd(ocTokenCommand)
+				backToCurrentUserCommand = fmt.Sprintf("%s -t %s", baseOdoLoginCommand, t)
+				testUserLoginCommand = fmt.Sprintf("%s -u %s -p %s", baseOdoLoginCommand, loginTestUser, loginTestUserPassword)
+				testUserCreateProject1Command = fmt.Sprintf("%s %s", baseOdoProjectCreate, odoTestProject1)
+				testUserLoginFailCommandWithToken = fmt.Sprintf("%s -t verybadtoken", baseOdoLoginCommand)
+			})
+		})
+
+		Context("Run login tests", func() {
+			AfterEach(func() {
+				projects := strings.Split(runCmd("oc projects -q"), "\n")
+				var waitOut bool
+				for _, p := range projects {
+					if len(p) > 0 {
+						waitOut = waitForCmdOut(fmt.Sprintf("%s %s", baseOdoProjectDelete, p), 10, func(out string) bool {
+							return strings.Contains(out, fmt.Sprintf("Deleted project : %s", p))
+						})
+						Expect(waitOut).To(BeTrue())
+					}
+				}
+				runCmd(backToCurrentUserCommand)
+			})
+
+			It("Should login successfully with username and password without any projects with appropriate message", func() {
+				session = runCmd(testUserLoginCommand)
+				Expect(session).To(ContainSubstring("Login successful"))
+				Expect(session).To(ContainSubstring("You don't have any projects. You can try to create a new project, by running"))
+				Expect(session).To(ContainSubstring("odo project create <project-name>"))
+				session = runCmd(ocWhoamiCommand)
+				Expect(session).To(ContainSubstring(loginTestUser))
+				token := runCmd(ocTokenCommand)
+				// One initialization needs one login, hence it happens here
+				testUserLoginCommandWithToken = fmt.Sprintf("%s -t %s", baseOdoLoginCommand, token)
+			})
+
+			It("Should login successfully with token without any projects with appropriate message", func() {
+				session = runCmd(testUserLoginCommandWithToken)
+				Expect(session).To(ContainSubstring("Logged into"))
+				Expect(session).To(ContainSubstring("You don't have any projects. You can try to create a new project, by running"))
+				Expect(session).To(ContainSubstring("odo project create <project-name>"))
+				session = runCmd(ocWhoamiCommand)
+				Expect(session).To(ContainSubstring(loginTestUser))
+			})
+
+			It("Should login successfully with username and password single project with appropriate message", func() {
+				runCmd(testUserLoginCommand)
+				runCmd(testUserCreateProject1Command)
+				runCmd(backToCurrentUserCommand)
+				session = runCmd(testUserLoginCommand)
+				Expect(session).To(ContainSubstring("Login successful"))
+				Expect(session).To(ContainSubstring(odoTestProject1))
+				session = runCmd(ocWhoamiCommand)
+				Expect(session).To(ContainSubstring(loginTestUser))
+			})
+
+			It("Should fail login on invalid token with appropriate message", func() {
+				session = runFailCmd(testUserLoginFailCommandWithToken, 1)
+				Expect(session).To(ContainSubstring("The token provided is invalid or expired"))
+				runCmd(testUserLoginCommand)
```

Comment:
What is the need of ```runCmd``` even after validating the error string. Don't think this is unnecessary for this scenario ?

Reply:
@amitkrout before entering this user is developer.  It attempts to login with invalid token,  so user is still developer. 

The clean up logic in AfterEach is supposed to run against testdeveloper user,  but will run against developer and may result in unexpected state for any further or future testing

Reply:
@mohammedzee1000 Thanks for the details